Write news to elesticsearch

oks29b opened the pull request.

Commit:
added rss to db

Commit:
added elasticsearch, writed repository layer and entity

Commit:
added search and saved news in elasticsearch

sergey-oreshkin commented on src/main/java/news/agregator/dto/NewsDto.java:
```diff
@@ -0,0 +1,15 @@
+public class NewsDto {
+    private String title;
+    private String description;
+    private Date localDate;
```
Date устарел, сейчас лучше использовать LocalDateTime

sergey-oreshkin commented on src/main/java/news/agregator/service/impl/NewsManagerImpl.java:
```diff
@@ -0,0 +1,103 @@
+    private final RssMapper rssMapper;
+    private final NewsRepository newsRepository;
+
+    @Async
```
по моему тут асинк ничего не даст

oks29b replied:
если время ставить меньше, может получаться так, что будет ожидаться окончание выполнения метода, а с @Async в параллельных потоках будет выполнение

sergey-oreshkin commented on src/main/java/news/agregator/service/impl/NewsManagerImpl.java:
```diff
@@ -0,0 +1,103 @@
+     * Считывает из db адреса rss и возвращает их как значения Map
+     */
+    private Map<String, String> getSources() {
+        List<RssDto> listRss = rssMapper.toListRssDto(rssEntityRepository.findAll());
```
зачем тут мапить в дто, можно сразу из энтити брать значения. и сделай это стримом.

sergey-oreshkin commented on src/main/java/news/agregator/service/impl/NewsManagerImpl.java:
```diff
@@ -0,0 +1,103 @@
+        for(String rss : sources.values()){
+            SyndFeed syndFeed = feedFromUrl(rss);
+            if(syndFeed == null) continue;
+            List<SyndEntry> entries = (List<SyndEntry>) syndFeed.getEntries();
```
вот это все надо вынести в отдельный сервис, чтоб можно было гибко менять источник новостей.
так же все сетевые запросы надо распараллелить, чтоб они выполнялись параллельно в несколько потоков.
вообще можно все запарралелить - получение, парсинг и сохранение в эластик по каждому источнику чтоб было в своем отдельном потоке парралельно.

Commit:
changed connectTimeout to socketTimeout and type veriable from Date to LocalDate

saprated funcionality for sources and take out elasticsearch connection properties in a application.yml

added asynchromius functionality

Commit:
added reading news with multithreading

Commit:
changed logic for multithreading reading news

Commit:
fixed checkstyle

Commit:
added githubaction

Commit:
add gradlew

Commit:
add gradle
